teku/src/main/java/tech/pegasys/teku/cli/options/LoggingOptions.java

```diff
@@ -160,4 +161,15 @@ public static String getDefaultLogFileGivenDataDir(String dataPath, boolean isVa
       return dataPath + SEP + "logs" + SEP + "teku.log";
     }
   }
+
+  public static String getLogPatternGivenDataDir(
+      final String dataPath, final String logFileNamePattern) {
+    if (logFileNamePattern.equals(DEFAULT_LOG_PATH_PATTERN)) {
+      return StringUtils.joinWith(SEP, dataPath, "logs", DEFAULT_LOG_FILE_NAME_PATTERN);
+    }
+    if (!logFileNamePattern.contains(SEP)) {
+      return StringUtils.joinWith(SEP, dataPath, "logs", logFileNamePattern);
+    }
+    return logFileNamePattern;
+  }
```

Comment:
So, turns out this should have taken into account whether it was the validator or not too.  Result should be the same as the default log file but with `_%d{yyyy-MM-dd}` inserted before the `.log` extension.